katdal/visdatav4.py

```diff
@@ -445,14 +446,11 @@ def _before(date):
                 corrected_weights = self._make_corrected(apply_weights_correction,
                                                          self.source.data.weights)
                 unscaled_weights = self.source.data.unscaled_weights
-                name = self.source.data.name
                 # Acknowledge that the applycal step is making the L1 product
-                if 'sdp_l0' in name:
-                    name = name.replace('sdp_l0', 'sdp_l1')
-                else:
-                    name = name + ' (corrected)'
+                if 'sdp_l0' in self.name:
```

Comment:
I would harvest the cal streams from `self.applycal_products` like `{cp.split('.')[0] for cp in self.applycal_products}`, then do two ifs:
```Python
if 'sdp_l0' in self.name and 'l1' in cal_streams:
    ...
if 'sdp_l1' in self.name and 'l2' in cal_streams:
    ...
```